### PR TITLE
Gem/taglist

### PR DESCRIPTION
**Pull request by Devyani606**
added gem for difficulty

### CODE REVIEW

File: app/controllers/api/v1/admin/problems_controller.rb [hunk 1]
```diff
@@ -19,20 +19,43 @@ def create
...
+                   (tag_result(tags) if params[:tag_list].present?) || (difficulty_result(difficulty_tag) if difficulty_tag) || Problem.kept
           problems = result.paginate(page: params[:page], per_page: params[:per_page]).order('id')
           render_success(data: { problems: serialize_resource(problems, ProblemSerializer),
-                                 page: problems.current_page, pages: problems.total_pages, total_problems: Problem.count },
+                                 page: problems.current_page, pages: problems.total_pages, total_problems: result.count },
                          message: I18n.t('index.success', model_name: Problem))
         end
 
+        def search_result(search)
+          return unless search && !search.blank?
```
Comment (somnath77) [inline app/controllers/api/v1/admin/problems_controller.rb:36]: You are validating 'search' here and up there at line 27, remove from line 27, just call function and don't validate there, applies for all functions

File: app/controllers/api/v1/admin/problems_controller.rb [hunk 2]
```diff
@@ -19,20 +19,43 @@ def create
...
+
+          Problem.kept.where('lower(title) LIKE ?', "%#{search.downcase}%")
+        end
+
+        def difficulty_result(_difficulty_tag)
```
Comment (somnath77) [inline app/controllers/api/v1/admin/problems_controller.rb:41]: taking _difficulty_tag param from function call but not using it

File: app/controllers/api/v1/admin/problems_controller.rb [hunk 3]
```diff
@@ -19,20 +19,43 @@ def create
...
+          return unless params[:difficulty_list]
+
+          Problem.tagged_with(params[:difficulty_list].downcase)
+        end
+
+        def tag_result(_tags)
```
Comment (somnath77) [inline app/controllers/api/v1/admin/problems_controller.rb:47]: same as above

File: app/controllers/api/v1/admin/problems_controller.rb [hunk 4]
```diff
@@ -19,20 +19,43 @@ def create
...
+          return unless params[:tag_list]
+
+          Problem.tagged_with(params[:tag_list].downcase)
+        end
+
+        def tag_and_difficulty_result(_tags, _difficulty_tag)
```
Comment (somnath77) [inline app/controllers/api/v1/admin/problems_controller.rb:53]: same as above

File: db/schema.rb [hunk 5]
```diff
@@ -123,6 +123,8 @@
     t.datetime 'updated_at', null: false
     t.integer 'submission_count'
     t.integer 'time_in_minutes', default: 60
+    t.string 'image'
+    t.boolean 'is_active', default: true
```
Comment (somnath77) [inline db/schema.rb:127]: Aare these changes necessary ?

Comment (Devyani606) [reply db/schema.rb:127]: no i wil remove it

File: app/controllers/api/v1/admin/problems_controller.rb (hunk 1 above)
Comment (Devyani606) [reply app/controllers/api/v1/admin/problems_controller.rb]: done

File: app/controllers/api/v1/admin/problems_controller.rb (hunk 2 above)
Comment (Devyani606) [reply app/controllers/api/v1/admin/problems_controller.rb]: done

File: db/schema.rb (hunk 5 above)
Comment (Devyani606) [reply db/schema.rb]: done

**Review by somnath77 — APPROVED** [review]

**Pull request merged**